### PR TITLE
Homework week3

### PR DESCRIPTION
**Pull request by Meazer**

### CODE REVIEW

File: Week3/homework/1-step3.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
 'use strict';
 
 function foo(func) {
-  // What to do here? 
+
+  console.log('Hello, I am foo! I can call another function:');
+  func();
+
```
Comment (rohni) [inline Week3/homework/1-step3.js]: Nice work! 
I would suggest returning the `func()` call.
The `foo` function does not know how the function will be used, and so should return the results of calling the function.
map, filter, and reduce all return the results of calling the function on the element.

File: Week3/homework/2-step3.js [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
-threeFive(10, 15, sayThree, sayFive);
+threeFive(10, 20, sayThree, sayFive);
+
+function sayThree(num) {
+  console.log(num + ' is divisible by three');
+}
+
+function sayFive(num) {
+  console.log(num + ' is divisible by five');
+}
```
Comment (rohni) [inline Week3/homework/2-step3.js]: ✅

File: Week3/homework/3-step3.js [hunk 3]
```diff
@@ -1,25 +1,47 @@
...
+  do {
+    repeatedStr += str;
+    num--;
+  } while (num > 0);
+
+  return repeatedStr;
 }
 
-console.log('while', repeatStringNumTimesWithDoWhile('abc', 3));
+console.log('do..while', repeatStringNumTimesWithDoWhile('abc', 3));
```
Comment (rohni) [inline Week3/homework/3-step3.js]: ✅

File: Week3/homework/4-step3.js [hunk 4]
```diff
@@ -1,2 +1,19 @@
...
+  this.wheels = 2;
+  this.engines = 1;
+  this.seats = 1;
+};
+
+const myCar = new Car();
+console.log(myCar);
+
+const myMotorBike = new MotorBike();
+console.log(myMotorBike);
```
Comment (rohni) [inline Week3/homework/4-step3.js]: ✅

File: Week3/homework/5-step3.js [hunk 5]
```diff
@@ -1,2 +1,19 @@
...
+    for (let j = 0; j < arr[i].length; j++) {
+      product *= arr[i][j];
+    }
+  }
+
+  return product;
+}
+
+
+console.log(multiplyAll([[1, 2], [3, 4], [5, 6, 7]]));
```
Comment (rohni) [inline Week3/homework/5-step3.js]: ✅

File: Week3/homework/6-step3.js [hunk 6]
```diff
@@ -3,4 +3,32 @@
...
 
-// add your solution here, or add a comment on how you would tackle this problem
+function printArray(arr) {
+
+  // if arr is an array call printArray for each element else print it.
+  if (Array.isArray(arr)) {
+    for (let i = 0; i < arr.length; i++) {
+      printArray(arr[i]);
+    }
+  } else console.log(arr);
```
Comment (rohni) [inline Week3/homework/6-step3.js]: Nice work.  
One style point here.  While what you wrote is legal and interpreted correctly by the interpretor, it is confusing to read.  I would suggest wrapping the else statement in {} as well.

**Comment by rohni** [comment]
Nice work Meazer.
Nicely laid out.  I have some comments, but mostly they are style or ideas of different ways to approach a particular problem that you might find interesting or useful.

Well done, and good luck with the rest of the course. 😄

**Comment by Meazer** [comment]
Thank you Rohni for your time

**Pull request closed without merging**